src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -322,20 +321,16 @@ private BaseResponseBean doSyncGetNext(FormplayerSyncScreen nextScreen,
     }
 
     private NotificationMessage doSync(FormplayerSyncScreen screen) throws Exception {
-        ResponseEntity<String> responseEntity = syncRequester.makeSyncRequest(screen.getUrl(),
-                screen.getBuiltQuery(),
-                restoreFactory.getUserHeaders());
-        if (responseEntity == null) {
-            return new NotificationMessage("Session error, expected sync block but didn't get one.", true, NotificationMessage.Tag.sync);
```

Comment:
Is the error that caused this message no longer possible?

Reply:
The response can never be null (there's code in RestTemplate to ensure that). In the case of errors an exception is raised so I don't think this case could ever have been triggered (certainly not under the current code configuration, perhaps previously if we were using the `errorPassthroughRestTemplate` but even then I think a `null` response is not allowed and a `IllegalStateException` would have been raised.